app/javascript/src/controllers/score_controller.js

```diff
@@ -13,6 +36,14 @@ export default class extends Controller {
     })
   }
 
+  /* Validate the current field, creating a popup if something is incorrect.
+   *
+   * For a field to be valid, it must contain one of the allowed values, 0 - 5
+   * (integers). Additionally, the score field but not exceed the goal field.
+   *
+   * This function makes very heavy use of field IDs to map between the current
```

Comment:
not sure the "very heavy use" is warranted here? there are two lines in this function which find a goal if `currentTarget` is a score and vice versa.